homeassistant/components/tradfri/__init__.py

```diff
@@ -87,6 +88,13 @@
         psk=entry.data[CONF_KEY],
         loop=hass.loop
     )
+
+    def on_hass_stop(event):
+        """Close connection when hass stops."""
+        hass.async_add_job(factory.shutdown())
```

Comment:
We don't use this function anymore. If it a coro use `async_create_task`. You need also decorate that as callback.

Reply:
@pvizeli hmm ok, I lifted the example from elsewhere in the code after searching for `async_listen_once(EVENT_HOMEASSISTANT_STOP`.